fontforge/python.c

```diff
@@ -14066,6 +14066,150 @@ return (NULL);
 return( Py_BuildValue("d",val));
 }
 
+static PyObject *PyFF_Font_get_style_set_names(PyFF_Font *self, void *UNUSED(closure)) {
+    int cnt, i;
+    SplineFont *sf = self->fv->sf;
+    struct otffeatname *fn;
+    struct otfname *on;
+    PyObject *ss_names_tuple, *ss_name_spec;
+    char *lang_str, tag_str[5];
+
+    if ( CheckIfFontClosed(self) )
+        return( NULL );
+
+    for ( cnt=0, fn=sf->feat_names; fn!=NULL; fn=fn->next )
+        for ( on=fn->names; on!=NULL; on=on->next, ++cnt );
+    if ( cnt==0 )
+        return( PyTuple_New(0) );
+    ss_names_tuple = PyTuple_New(cnt);
+    for ( cnt=0, fn=sf->feat_names; fn!=NULL; fn=fn->next ) {
+        for ( on=fn->names; on!=NULL; on=on->next, ++cnt ) {
+            lang_str = copy(NOUI_MSLangString(on->lang));
+            ss_name_spec = PyTuple_Pack(3, PyUnicode_FromString(lang_str), TagToPythonString(fn->tag, false), PyUnicode_FromString(on->name));
```

Comment:
Doesn't `PyUnicode_FromString()` leave its argument unaffected (e.g. the string is copied into a different form rather than referenced)? If so, aren't we leaking the copy() in the previous line?